compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/utils.kt

```diff
@@ -484,3 +487,140 @@ private val PropertyDescriptor.declaresDefaultValue: Boolean
         is KtParameter -> declaration.defaultValue != null
         else -> false
     }
+
+fun <T : Annotation> KSAnnotated.getAnnotation(annotationKClass: KClass<T>): T? {
+    return this.annotations.firstOrNull {
+        annotationKClass.qualifiedName == it.annotationType.resolve().declaration.qualifiedName?.asString()
+    }
+        ?.toAnnotation(annotationKClass)
+}
+
+fun KSAnnotated.getAnnotations(): Sequence<out Annotation> {
+    return this.annotations.map {
+        val kClass = Class.forName(it.annotationType.resolve().declaration.qualifiedName?.asString()).kotlin
+        it.toAnnotation(kClass as KClass<Annotation>)
+    }
+}
+
+fun <T : Annotation> KSAnnotated.getAnnotationsByType(annotationKClass: KClass<T>): Sequence<T> {
+    return this.annotations.filter {
+        it.annotationType.resolve().declaration.qualifiedName?.asString() == annotationKClass.qualifiedName
+    }.map { it.toAnnotation(annotationKClass) }
+}
+
+fun <T : Annotation> KSAnnotated.isAnnotationPresent(annotationKClass: KClass<T>): Boolean =
+    getAnnotation(annotationKClass)?.let { true } ?: false
+
+private fun <T : Annotation> KSAnnotation.toAnnotation(annotationKClass: KClass<T>): T {
+    val clazz = annotationKClass.java
+    @Suppress("UNCHECKED_CAST")
+    return Proxy.newProxyInstance(clazz.classLoader, arrayOf(clazz), createInvocationHandler(clazz)) as T
+}
+
+@Suppress("TooGenericExceptionCaught")
+private fun KSAnnotation.createInvocationHandler(clazz: Class<*>): InvocationHandler {
+    val cache = ConcurrentHashMap<Pair<Class<*>, Any>, Any>(arguments.size)
+    return InvocationHandler { proxy, method, _ ->
+        if (method.name == "toString" && arguments.none { it.name?.asString() == "toString" }) {
+            "${clazz.canonicalName}" +
+                arguments.map { argument: KSValueArgument ->
+                    // handles default values for enums otherwise returns null
+                    val methodName = argument.name?.asString()
+                    val value = proxy.javaClass.methods.find { m -> m.name == methodName }?.invoke(proxy)
+                    "$methodName=$value"
+                }.toList()
```

Comment:
This handles the `toString` method and obtaining the default values by reusing the implemented proxy methods